tests/tests/swfs/avm1/form_loader_encoding_1/test.as

```diff
@@ -0,0 +1,18 @@
+// SWF Version 6
+
+/*
+ * This test is the first test out of a series of tests, testing the encoding the form loader uses to decode
+ * text files.
+ * This test tests whether a UTF-8 text file loaded in an SWF file with SWF version 6 is decoded as UTF-8.
+ * Flash decodes invalid UTF-8 characters (such as some characters encoded in ISO Latin-1) differently, and
+ * as of June 2024, Ruffle doesn't implement this. Therefore, this test only tests a UTF-8 file.
```

Comment:
I would suggest adding those files anyway, commenting out the code that verifies their values, and adding a TODO. Alternatively a 4th version of this test that is a known failure?

Reply:
I think adding a new test just for this is a bit overkill.
Essentially, this means that when Flash loads a file with invalid UTF-8 characters, it decodes them as some seemingly random symbols while Ruffle just only displays the valid parts.
Realistically, that behaviour isn't really wanted (if you want any symbols, you can just put them as UTF-8 in a file, this mostly happens when a file is wrongly encoded).

I can add an explicit TODO as you suggested. But I'm not completely sure if we want to implement this; it's likely not very easy. I can imagine that we'd need to write an own UTF-8 parser, and going through every text file like this would also include a slight (although probably insignificant) performance penalty. Because of this, I just wrote it as a notice.

Reply:
The truth is that every observable behavior may be depended upon. There are some obfuscated SWFs that rely on decoding invalid UTF-8 (e.g. https://github.com/ruffle-rs/ruffle/issues/15919), or some SWFs currently panic in Ruffle due to decoding errors (e.g. https://github.com/ruffle-rs/ruffle/pull/12221, I myself have encountered some).

I was thinking in terms of:
1. some SWFs may depend on this behavior,
2. this behavior may help resolve similar issues related to decoding invalid UTF-8.

TBH I'm not sure how exactly they are related, just noticed that all of them have decoding invalid UTF-8 in common :)

Reply:
Okay, that's actually pretty reasonable. Thanks for the explanation.
What do you think would be the best approach?

Reply:
I would personally suggest adding a 4th test that is a known failure and describing the issue there, but a TODO comment with commented out code is also okay if you don't want to add another test :)

If you have any observations etc. related to FP in this case it would also be great to include them in comments!

Reply:
Done :)